lib/SILOptimizer/Transforms/StringSwitch.cpp

```diff
@@ -0,0 +1,336 @@
+      return false;
+
+    results.pop_back();
+    builder.getASTContext().lookupInSwiftModule("_allocateUninitializedArray",
+                                                results);
+    assert(results.size());
+    if (results.size() != 1)
+      return false;
+
+    funcDecl = dyn_cast<FuncDecl>(results.front());
+    if (!funcDecl)
+      return false;
+
+    funcDeclRef = SILDeclRef(funcDecl, SILDeclRef::Kind::Func);
+    // TODO: this only exists when there is an enum in the source file.
+    makeArray = funcBuilder.getOrCreateFunction(
+        dummyLoc, funcDeclRef, ForDefinition_t::NotForDefinition);
+    if (!makeArray)
+      return false;
+
+    results.pop_back();
+    // Look up "assert" to get the StaticString type.
+    builder.getASTContext().lookupInSwiftModule("assert", results);
+    assert(results.size());
+    if (results.size() != 1)
+      return false;
+
+    funcDecl = dyn_cast<FuncDecl>(results.front());
+    if (!funcDecl)
+      return false;
+
+    funcDeclRef = SILDeclRef(funcDecl, SILDeclRef::Kind::Func);
+    SILFunction *assertFn = funcBuilder.getOrCreateFunction(
+        dummyLoc, funcDeclRef, ForDefinition_t::NotForDefinition);
+    if (!assertFn)
+      return false;
+
+    // General variables we'll need later.
+    wordTy = SILType::getBuiltinWordType(builder.getASTContext());
+    byteTy = SILType::getBuiltinIntegerType(8, builder.getASTContext());
+    boolTy = SILType::getBuiltinIntegerType(1, builder.getASTContext());
+    i64Ty = SILType::getBuiltinIntegerType(64, builder.getASTContext());
+    assertFnFuncTy = assertFn->getLoweredFunctionType();
+    staticStringBaseTy = assertFnFuncTy->getParameters()[2].getSILStorageType(
+        module, assertFnFuncTy);
+    staticStringTy = staticStringBaseTy.getObjectType();
+    staticStringAddrTy = staticStringBaseTy.getAddressType();
+    replacementFuncTy = replacementFunc->getLoweredFunctionType();
+    staticStringArrayTy = replacementFuncTy->getParameters()[0]
+                              .getSILStorageType(module, replacementFuncTy)
+                              .getAddressType();
+    boolObjTy = SILType::getPrimitiveObjectType(builder.getASTContext()
+                                                    .getBoolDecl()
+                                                    ->getDeclaredType()
+                                                    ->getCanonicalType());
+    return true;
+  }
+
+public:
+  StringSwitchPass() = default;
+
+  /// The entry point to the transformation.
+  void run() override {
+    if (!initialize())
+      return;
+
+    // A vector of string literals and string compares that are associated with
+    // a single subject stirng.
+    struct CompareSet {
+      SmallVector<StringLiteralInst *, 12> stringLiterals;
+      SmallVector<ApplyInst *, 12> stringCompares;
+    };
+
+    // Map a string to a vector of compares and literals.
+    llvm::MapVector<SILValue, SmallVector<CompareSet, 1>> worklist;
```

Comment:
Can you change the name `worklist` to something else. `worklist` is kind of a "reserved" name for something where you constantly push _and_ pop things.